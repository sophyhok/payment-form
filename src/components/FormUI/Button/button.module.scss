.btn {
  border: 1px solid transparent;
  line-height: 32px;
  border-radius: 5px;
  cursor: pointer;

  &__primary {
    background: #2C76DC;
    border-color:#2C76DC;
    color: #FFFFFF;
  }

  &__primaryInverse {
    background: #FFFFFF;
    border-color:#FFFFFF;
    color:  #2C76DC;
  }
}