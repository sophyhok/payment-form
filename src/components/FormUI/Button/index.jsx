import React from 'react';
import { string, oneOf, func } from 'prop-types';
import classnames from 'classnames';

import styles from './button.module.scss';

const propTypes = {
  label: string,
  onClick: func,
  variant: oneOf([
    'primary',
    'primaryInverse',
  ]),
  testId: string,
}

const defaultProps = {
  label: 'button',
  onClick: null,
  variant: 'primary',
  testId: 'button',
}

const Button = ({ label, onClick, variant, testId }) => {
  return (
    <button
      className={classnames(styles.btn, styles[`btn__${variant}`])}
      onClick={onClick}
      data-testid={testId}
    >
      {label}
    </button>
  )
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;