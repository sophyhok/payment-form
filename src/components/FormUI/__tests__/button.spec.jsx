import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Button from '../Button';

const onClick = jest.fn();

it('should match snapshot', () => {
  const { asFragment } = render(<Button />)
  expect(asFragment()).toMatchSnapshot();
});

it('should call the onClick function when the button is clicked', () => {
  const { getByText } = render(<Button onClick={onClick} />);
  const button = getByText('button');
  fireEvent.click(button);

  expect(onClick).toHaveBeenCalled();
});

it('should render primaryInverse variant', () => {
  const { getByText } = render(
    <Button variant="primaryInverse"/>,
  );
  const button = getByText('button');
  expect(button).toHaveClass('btn__primaryInverse');
});