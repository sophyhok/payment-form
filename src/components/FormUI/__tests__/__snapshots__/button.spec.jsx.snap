// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn__primary"
    data-testid="button"
  >
    button
  </button>
</DocumentFragment>
`;
